src/cli/cli_udp.cpp

```diff
@@ -172,17 +157,65 @@ otError UdpExample::ProcessSend(int argc, char *argv[])
         messageInfo.mPeerPort = static_cast<uint16_t>(value);
     }
 
+    if (argc == 2 || argc == 4)
+    {
+        int typePos = curArg++;
+
+        if (strcmp(argv[typePos], "-s") == 0)
+        {
+            unsigned long value;
+            payloadType = kTypeAutoSize;
+            SuccessOrExit(error = Interpreter::ParseUnsignedLong(argv[curArg], value));
```

Comment:
when `argc == 3`, this code will access the end of `argv`, i.e. "\0", and ParseUnsignedLong("\0") will return OT_ERROR_NONE with value=0, which I think is also a minor issue.

Reply:
I don't think it will happen as this block has the condition of `argc == 2 || argc == 4`.